src/cdpy/df.py

```diff
@@ -7,48 +7,69 @@ class CdpyDf(CdpSdkBase):
+            services = self.list_services(env_crn=env_crn)
+            if len(services) == 0:
+                return None
+            elif len(services) == 1:
+                resolved_df_crn = services[0]['crn']
+            else:
+                self.sdk.throw_error(
+                    CdpError('More than one DataFlow service found for env_crn, please try list instead')
+                )
+        else:
+            self.sdk.throw_error(CdpError("Either df_crn or env_crn must be supplied to df.describe_service"))
         return self.sdk.call(
-            svc='df', func='get_environment', ret_field='environment', squelch=[
+            svc='df', func='describe_service', ret_field='service', squelch=[
                 Squelch(value='NOT_FOUND',
-                        warning='No Environment with crn %s for DataFlow found in Tenant' % env_crn)
+                        warning='No DataFlow Deployment with crn %s found' % df_crn),
+                Squelch(value='PERMISSION_DENIED')  # DF GRPC sometimes returns 403 when finishing deletion
             ],
-            crn=env_crn
+            serviceCrn=resolved_df_crn
         )
 
-    def enable_environment(self, env_crn: str, authorized_ips: list = None, min_nodes: int = 3, max_nodes: int = 3,
-                           enable_public_ip: bool = True):
+    def enable_service(self, env_crn: str, lb_ips: list = None, min_nodes: int = 3, max_nodes: int = 3,
+                       enable_public_ip: bool = True, kube_ips: list = None, cluster_subnets: list = None,
+                       lb_subnets: list = None):
         self.sdk.validate_crn(env_crn)
         return self.sdk.call(
-            svc='df', func='enable_environment', ret_field='environment',
-            crn=env_crn, minK8sNodeCount=min_nodes, maxK8sNodeCount=max_nodes,
-            usePublicLoadBalancer=enable_public_ip, authorizedIpRanges=authorized_ips
+            svc='df', func='enable_service', ret_field='service',
+            environmentCrn=env_crn, minK8sNodeCount=min_nodes, maxK8sNodeCount=max_nodes,
+            usePublicLoadBalancer=enable_public_ip, kubeApiAuthorizedIpRanges=kube_ips,
+            loadBalancerAuthorizedIpRanges=lb_ips, clusterSubnets=cluster_subnets, loadBalancerSubnets=lb_subnets
         )
 
-    def disable_environment(self, env_crn: str, persist: bool = False):
-        self.sdk.validate_crn(env_crn)
+    def disable_service(self, df_crn: str, persist: bool = False, terminate=False):
+        self.sdk.validate_crn(df_crn)
         return self.sdk.call(
-            svc='df', func='disable_environment', ret_field='status', ret_error=True,
-            crn=env_crn, persist=persist
+            svc='df', func='disable_service', ret_field='status', ret_error=True,
+            serviceCrn=df_crn, persist=persist, terminateDeployments=terminate
         )
 
-    def delete_environment(self, env_crn: str):
-        self.sdk.validate_crn(env_crn)
+    def reset_service(self, df_crn: str):
```

Comment:
My for my own education, what does this do? The CLI "help" for this command tells me very little.